joomla/joomla-cms
# [4.0] Converting additional mails to mail templates

Pull Request for Issue #29215.

### Summary of Changes
This converts the mails of the actionlogs component, the privacy component and the privacy consent plugin to the mail template system.

* [X]  Notification mail from com_actionlogs (administrator\components\com_actionlogs\src\Model\ActionlogModel.php line 174)
* [X]  com_messages notification (administrator\components\com_messages\src\Model\MessageModel.php line 435)
* [X]  com_privacy send export to user (administrator\components\com_privacy\src\Model\ExportModel.php line 240)
* [X]  com_privacy notification about a new request (administrator\components\com_privacy\src\Model\RequestModel.php line 362)
* [X]  com_privacy creating a new request (components\com_privacy\src\Model\RequestModel.php line 188/190 ATTENTION!! HERE IS A BUG! The mail is send twice!)
* [X]  reminder of expiring privacy consent (plugins\system\privacyconsent\privacyconsent.php line 604)
* [X]  User registration (components\com_users\src\Model\RegistrationModel.php line 198, 254, 650, 711)

The mentioned bug has also been fixed.

### Testing Instructions
You have to trigger the feature to send these mails to test this.
- Go to your user profile in the backend and enable to receive actionlog updates. You should get a mail when you login.
- Create a request for export or removal in the frontend privacy component to trigger the frontend mails.
- Create a request for export or removal in the backend privacy component to trigger the backend mails.
- Send the export of data in the backend via mail to trigger that one...
- Enable renewable consents, give consent and then edit it in the database to be close to expiration. That should trigger that last mail.
- Register users with all the different settings we have.
